### ISSUE TITLE
self-signed cert error when downloading browser form artifact repository

### ISSUE THREAD

**Issue by brendonts** [microsoft/playwright-python#1706]
**Context:**
- Operating System: Linux
- Python version: 3.x
- Browser: Chromium
- Extra: pulling from Artifact repo

**Describe the bug**

I am trying to install chromium inside my python virtual env from an artifact repository but am receiving "Error: self signed certificate in certificate chain"
Running python3 reqests.get(https://my.artifact.repo) initially returned the same SSL error which made sense because my OS certs were not in my pyvenv, however I added our certs to the pyvenv/lib64/python3.x/site-packages/certifi/mycacerts.pem and re-running the get request using requests returned: <Response [200]> so I believe the certs are working as expected (artifact repo via browser works as expected too).  

However, when I run the following:
```
PLAYWRIGHT_DOWNLOAD_HOST=HTTPS://my.artifact.repository/././chromiumplaywright install
```
I still receive:
```
"Error: self signed certificate in certificate chain"
Error: Download failure, code=1, at ChildProcess.<anonymous> (/pyvenv/lib/python3.x/site-packages/playwright/driver/package/lib/server/registry/BrowserFetcher.js
```

I'm a bit ignorant to how the Python version of Playwright is running a Javascript file but I did follow the [Playwright documentation for "Browsers>Download from an artifact repository"](https://playwright.dev/python/docs/browsers) and even tried adding in the:
```
NODE_EXTRA_CA_CERTS=pyvenv/lib64/python3.x/site-packages/certifi/mycacerts.pem
```
environment variable to specify my cert path but received the same self-signed cert error.  

I did go on a Node.js tangent and some of the documentation seems to imply that NODE_EXTRA_CA_CERTS can only be set when the initial Node.js process is first launches: https://nodejs.org/api/cli.html#node_extra_ca_certsfile.  
This does make me wonder if I need to set the NODE_EXTRA_CA_CERTS as a variable in one of the .js files Playwright uses, but again I am a little ensure how these work.  Thank you.

**Comment by mxschmitt** [comment]
Playwright for Python uses Node.js under the hood, which we call the Playwright driver. That's why for such cases you need to set specific Node.js env vars.

Can you try passing an absolute path to it?

**Comment by brendonts** [comment]
Thank you and sorry I should have specified in my testing I used the absolute path just like so:
```
NODE_EXTRA_CA_CERTS=/home/pyvenv/lib64/python3.x/site-packages/certifi/mycacerts.pem
and
NODE_EXTRA_CA_CERTS="/home/pyvenv/lib64/python3.x/site-packages/certifi/mycacerts.pem"
```
I saw other closed issues where folks had success using NODE_EXTRA_CA_CERTS but I'm not sure node is picking up the env variable set via linux cli. The Node documentation I linked specified:
```
This environment variable is ignored when node runs as setuid root or has Linux file capabilities set.
The NODE_EXTRA_CA_CERTS environment variable is only read when the Node.js process is first launched. Changing the value at runtime using process.env.NODE_EXTRA_CA_CERTS has no effect on the current process.
```
So I wasn't sure if this environment var needed to be set somewhere in the Playwright driver.

**Comment by mxschmitt** [comment]
I think you are setting the wrong certificates, since we need the CA certs which trust your custom artifactory https server. certifi are just normal root certs which can be used by anybody from my understanding.

**Comment by brendonts** [comment]
So I figured out my issue more or less. Earlier I had replaced /certifi/mycacerts.pem with a copy of my certificate bundle so I could test it with python requests/certifi which I successfully did. Because that worked, I assumed this bundle would work with node.js and Playwright but for whatever reason it doesn't like the bundle when I feed it to NODE_EXTRA_CA_CERTS.  

Instead I saved a copy of just the CA cert from site within my browser as a .pem and tried that with NODE_EXTRA_CA_CERTS and am not longer having the certificate issue.

Thank you!
